### ISSUE TITLE
Error setting up WRF domains

### ISSUE THREAD

**Issue by putland** [GIS4WRF/gis4wrf#149]
Hi,

I have installed GIS4WRF on QGIS on Ubuntu 18.04 and trying to set up the WRF domains I get the following error.

Any help would be appreciated.

Thanks

An error has occurred while executing Python code: 

AssertionError 
Traceback (most recent call last):
  File "/home/user/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/ui/widget_domains.py", line 313, in on_set_projection_button_clicked
    qgsProject.setCrs(get_qgis_crs(crs.proj4))
  File "/home/user/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/geo.py", line 28, in get_qgis_crs
    assert srs_id != -1
AssertionError

Python version: 3.6.8 (default, Oct  7 2019, 12:59:55) [GCC 8.3.0] 
QGIS version: 3.10.0-A Coruña A Coruña, 6c816b4204 

Python Path:
/usr/share/qgis/python
/home/user/.local/share/QGIS/QGIS3/profiles/default/python
/home/user/.local/share/QGIS/QGIS3/profiles/default/python/plugins
/usr/share/qgis/python/plugins
/usr/lib/python36.zip
/usr/lib/python3.6
/usr/lib/python3.6/lib-dynload
/home/user/.local/lib/python3.6/site-packages
/usr/local/lib/python3.6/dist-packages
/usr/lib/python3/dist-packages
/home/user/.local/share/QGIS/QGIS3/profiles/default/python
/home/user/.local/share/gis4wrf/python36/local/lib/python3.6/dist-packages
/home/user/.local/share/gis4wrf/python36/lib/python3/dist-packages
/home/user/.local/share/gis4wrf/python36/lib/python3.6/dist-packages

**Comment by dmey** [comment]
What configuration and projection are you trying to set-up? Have you been able to run the tutorial at https://gis4wrf.github.io/tutorials/gis4wrf/simulate-the-2018-european-heat-wave-with-gis4wrf/? If not, please try that first and let us know.

**Comment by letmaik** [comment]
@putland Can you provide more details please?

**Comment by jhbravo** [comment]
I have this problem too, SO: Debian 10, QGIS version | 3.10.0-A Coruña

> 2019-11-18T11:59:16     WARNING    Traceback (most recent call last):
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/ui/tab_simulation.py", line 106, in open_project_from_object
              self.set_project_in_tabs()
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/ui/tab_simulation.py", line 73, in set_project_in_tabs
              self.domain_tab.project = project
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/ui/widget_domains.py", line 230, in project
              self.populate_ui_from_project()
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/ui/widget_domains.py", line 286, in populate_ui_from_project
              self.draw_bbox_and_grids(zoom_out=True)
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/ui/widget_domains.py", line 561, in draw_bbox_and_grids
              update_domain_outline_layers(self.iface.mapCanvas(), project, zoom_out=zoom_out)
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/geo.py", line 43, in update_domain_outline_layers
              qgs_crs = get_qgis_crs(proj4)
              File "/home/jorgeb/.local/share/QGIS/QGIS3/profiles/default/python/plugins/gis4wrf/plugin/geo.py", line 28, in get_qgis_crs
              assert srs_id != -1
             AssertionError

I had used GIS4WRF in this same machine, only that I reinstalled the OS and now I get that error

**Comment by letmaik** [comment]
Without more details we can't help you there. Please let us know if you can follow the tutorial. If you get stuck there already, then it's easier for us to reproduce. If it's with custom settings, then we need to know those to diagnose the issue.

**Comment by jhbravo** [comment]
Hi @letmaik thanks for your quick response
My interest is only to create the domains for the execution of the model, I already have my own methodology for this. But I really liked the graphical interface of GIS4WRF and how it can be adjusted very precisely to the area of interest, so what I do is the following:

1. Simulation → General → Open Project (the one generated by default)
2. Domain → MapType → GCS / Projection (Latitude / Longitude)

Here on the map I approach my area of interest and then press the (Ser Map CRS) button and that is where the error is

**Comment by letmaik** [comment]
I installed the latest QGIS version 3.10 and also get errors, though different ones, but also related to CRS. I think this is because QGIS uses the new GDAL 3.0 release which has [backwards incompatible changes](https://github.com/OSGeo/gdal/blob/release/3.0/gdal/MIGRATION_GUIDE.TXT). To unblock you, I suggest you use an older QGIS release, until we got around to adapt to the new version.

**Comment by letmaik** [comment]
@jhbravo @putland I fixed the issue that I saw myself but that's a different one than what you're seeing. Would you mind going to QGIS -> Help -> About and let me know the GDAL versions that are displayed there? (left and right column)

**Comment by jhbravo** [comment]
@letmaik my QGIS is in spanish



-- | --
Revisión del código de QGIS | 6c816b4204
Ejecutándose contra Qt | 5.11.3
Ejecutándose contra GDAL/OGR | 2.4.0
Ejecutándose contra GEOS | 3.7.1-CAPI-1.11.1 27a5e771
Running against SQLite | 3.27.2
Versión de SpatiaLite | 4.3.0a
Versión de QScintilla2 | 2.10.4
Ejecutándose contra PROJ | Rel. 5.2.0, September 15th, 2018



| First Header  | Second Header |
| ------------- | ------------- |
| Versión de QGIS : 3.10.0-A Coruña  | Content Cell  |
| Compilado contra Qt : 5.11.3 | Content Cell  |
|  |   |
|  |   |
|  |   |
|  |   |
|  |   |
|  |   |
|  |   |
Compilado contra GDAL/OGR : 2.4.0
Compilado contra GEOS : 3.7.1-CAPI-1.11.1
Compiled against : 3.27.2
Versión del cliente PostgreSQL : 11.5 (Debian 11.5-1+deb10u1)
Versión de QWT : 6.1.4
Compilado contra PROJ : 5.2.0
Versión del SO : Debian GNU/Linux 10 (buster)

**Comment by letmaik** [comment]
To diagnose this issue further, could you run QGIS with debug output enabled and upload the log file here (by drag and drop into comment field)? You can do that by launching qgis with extra environment variables set:
```sh
QGIS_DEBUG=9 QGIS_LOG_FILE=/tmp/qgis.log qgis
```

**Comment by jhbravo** [comment]
I have run QGIS with the environment variables and no qgis.log is generated
already change the file path and not generate any qgis.log
also modify it from Configuration -> Option -> System

**Comment by letmaik** [comment]
If you just set `QGIS_DEBUG=9` (not via the options but from the terminal), do you see any log output in the terminal when QGIS runs?

**Comment by jhbravo** [comment]
`QGIS_DEBUG=9 QGIS_LOG_FILE=/tmp/qgis.log qgis`
output:
```
Warning 4: Failed to open /usr/share/qgis/resources/data/world_map.gpkg: Permiso denegado.
Warning 4: Failed to open /usr/share/qgis/resources/data/world_map.gpkg: Permiso denegado.
Warning: QMainWindow::saveState(): 'objectName' not set for QDockWidget 0x5647ba76add0 'GIS4WRF;
```
when open QGIS I used GIS4WRF and only the lines above was written, the last line just when close qgis

**Comment by gsever** [comment]
Hello, I got the same error on Windows using the latest version of QGIS.

QGIS version | 3.10.0-A Coruña | QGIS code revision | 6ffa89eb3e
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.0.2 | Running against GDAL/OGR | 3.0.2
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.2.1 | Running against PROJ | Rel. 6.2.1, November 1st, 2019
OS Version | Windows 7 SP 1 (6.1)
Active python plugins | db_manager; MetaSearch; processing; gis4wrf


It shows the following error upon clicking 'Set Map CRS' button in Simulation/Domain tab.

`2019-11-21T15:00:10     WARNING    Traceback (most recent call last):
              File "C:/Users/gs/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_domains.py", line 313, in on_set_projection_button_clicked
              qgsProject.setCrs(get_qgis_crs(crs.proj4))
              File "C:/Users/gs/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\geo.py", line 28, in get_qgis_crs
              assert srs_id != -1
             AssertionError`

Previous versions 3.8 and 3.4 showed the same errors. Is there any recommended version for Windows?

**Comment by letmaik** [comment]
@gsever I'm on Windows and haven't seen this error yet.

Could someone try the following variant of the plugin by using the Install from ZIP option in the plugin manager? I don't expect it to work but there should be more detail in the error message.
[gis4wrf.zip](https://github.com/GIS4WRF/gis4wrf/files/3876489/gis4wrf.zip)

**Comment by dmey** [comment]
I did not have any issues on Windows 10 with GIS4WRF 0.14.1 and QGIS 3.8. 3 or 3.10.0 -- see below. I noticed that @gsever is using Windows 7 but I would not expect that to make a difference here. 

![image](https://user-images.githubusercontent.com/26348635/69373071-e71a9f80-0c9a-11ea-9bda-0dc72022e233.png)

![image](https://user-images.githubusercontent.com/26348635/69373471-c3a42480-0c9b-11ea-8c51-d08942a7f373.png)

without a proper log is rather difficult to diagnose this.

**Comment by gsever** [comment]
Following shows the output when I test using the zip file provided by @letmaik 

`2019-11-21T23:57:34     WARNING    Traceback (most recent call last):
              File "C:/Users/gs/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_domains.py", line 313, in on_set_projection_button_clicked
              qgsProject.setCrs(get_qgis_crs(crs.proj4))
              File "C:/Users/gs/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\geo.py", line 29, in get_qgis_crs
              assert srs_id != -1, proj4
             AssertionError: +proj=lcc +lat_1=30.0 +lat_2=60.0 +lat_0=0.0 +lon_0=-74.0 +x_0=0 +y_0=0 +datum=WGS84 +no_defs`


and yes I am on Windows 7. I wonder if other installed libraries are interfering with the QGIS..

**Comment by letmaik** [comment]
@gsever Weird, looks all normal. There is nice tool from Microsoft where you can see all debug messages from the system, including QGIS. It may be worth a try: https://docs.microsoft.com/en-us/sysinternals/downloads/debugview Just launch it and the messages will display, if there are any.

**Comment by gsever** [comment]
This issue is more like a Python error to me. Is it normal to get AssertionError raised when Set Map CRS button clicked? Perhaps I can test with a working "crs" line in geo.py to see if this is something to do with my settings. There wasn't any messages on debugview originating from QGIS.

**Comment by dmey** [comment]
@gsever we have been working on the assumption that you could not recreate the configuration on the tutorial (see my original question https://github.com/GIS4WRF/gis4wrf/issues/149#issuecomment-549707611). I am confused as it now seems that you are using a different configuration. Can you please confirm whether you get an error when using the following configuration (i.e. GCS/Projection: `Lambert Conformal`, True Latitude 1: `3.5`, True Latitude 2: `7`, Standard Longitude: `4`).  If you don't please let us know your configuration so that we can reproduce the issue on our side.

Note to self and @letmaik -- this exception can be easily reproduced, for example, by setting all values to 0 when using Lambert Conformal. This is expected however the exception message could be made a bit more informal.

**Comment by letmaik** [comment]
@dmey The error you observe with your invalid CRS (all 0s) is different, and yes, we can handle that better. The modified plugin package I offered above first checks if the CRS is valid and if so, adds it to the User CRSs if it doesn't exist already (this is a technical thing we have to do in order to "use" a custom CRS). With your invalid CRS you get the error 
```
...
assert crs.isValid(), proj4
AssertionError: +proj=lcc +lat_1=0.0 +lat_2=0.0 +lat_0=0.0 +lon_0=0.0 +x_0=0 +y_0=0 +datum=WGS84 +no_defs
```
Note the line of the assert where it fails: `crs.isValid()`.
In @gsever's case, it fails when checking `assert srs_id != -1` which means that the CRS was valid but it couldn't be inserted into QGIS's database for some reason, and it's exactly that error that everyone else is seeing except us. Note that I included the projection string in the assert message (in the modified package) for exactly the purpose of checking whether it looks abnormal, but @gsever's settings seem fine and I used the same locally and didn't get an error: `+proj=lcc +lat_1=30.0 +lat_2=60.0 +lat_0=0.0 +lon_0=-74.0 +x_0=0 +y_0=0 +datum=WGS84 +no_defs`.
In summary, I'm still confused.

**Comment by dmey** [comment]
@letmaik OK -- thanks for pointing this out.

I think the easiest would be to test this on latest Ubuntu or Debian 10 as others have already reported --but will unlikely have time to look at this until next week. The one on Windows 7 is the most confusing to me though.

**Comment by letmaik** [comment]
I found out that QGIS only prints debug messages if you install a development build.
@gsever For Windows it's quite easy, just install the latest from https://qgis.org/downloads/weekly/?C=M;O=D and then start DebugView to see detailed messages.
@jhbravo If you can install the weekly/nightly build somehow then I'm sure the environment variables have an effect or probably you don't have to set them and see everything on the terminal by default.
Hopefully that gives us a clue. In my case, the log looks like that:
```
00002333	9.31543064	[21264] ..\..\..\src\core\qgsmessagelog.cpp(29) : (QgsMessageLog::logMessage) [48863ms] 2019-11-22T09:35:30 CRS[1] Saved user CRS [+proj=lcc +lat_1=1.0 +lat_2=3.0 +lat_0=0.0 +lon_0=3.0 +x_0=0 +y_0=0 +datum=WGS84 +no_defs]	
...
```
Note that I disabled the background map (simply deselect the Stamen layer that gets added when you start GIS4WRF) to reduce debug noise.

**Comment by gsever** [comment]
That error was occurring for any combination of settings in the Map Type section when Lambert Comformal is selected. Interestingly, following @letmaik 's installing a recent weekly built (QGIS-OSGeo4W-3.11.0-46-Setup-x86_64) the issue goes away. I was ready to start DebugView and report back here, however I wasn't able to reproduce the error anymore. Maps shown fine and surrounding boxes are drawn based on configuration. You mentioned the issue isn't reproducible on Windows 10.  So perhaps this issue is Windows 7 specific? Anyhow, thanks for your helpful comments.

**Comment by noitul** [comment]
Dear all,
I have the same issues on Windows 10 and CentOS7. I tried to install GIS4WRF with qgis3.0, 3.4, 3.10, and 3.11 which is from the link above.  But I was not able to run the tutorial.
The error code was:

An error has occurred while executing Python code: 

AssertionError 
Traceback (most recent call last):
  File "C:/Users/SI/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_domains.py", line 313, in on_set_projection_button_clicked
    qgsProject.setCrs(get_qgis_crs(crs.proj4))
  File "C:/Users/SI/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\geo.py", line 28, in get_qgis_crs
    assert srs_id != -1
AssertionError


My system is:


QGIS version
3.10.0-A Coruña
QGIS code revision
6ffa89eb3e
Compiled against Qt
5.11.2
Running against Qt
5.11.2
Compiled against GDAL/OGR
3.0.2
Running against GDAL/OGR
3.0.2
Compiled against GEOS
3.8.0-CAPI-1.13.1
Running against GEOS
3.8.0-CAPI-1.13.1 
Compiled against SQLite
3.29.0
Running against SQLite
3.29.0
PostgreSQL Client Version
11.5
SpatiaLite Version
4.3.0
QWT Version
6.1.3
QScintilla2 Version
2.10.8
Compiled against PROJ
6.2.1
Running against PROJ
Rel. 6.2.1, November 1st, 2019
OS Version
Windows 10 (10.0)
Active python plugins
db_manager; 
MetaSearch; 
processing; 
gis4wrf

**Comment by letmaik** [comment]
@noitul Could you copy any relevant logs from [DebugView](https://docs.microsoft.com/en-us/sysinternals/downloads/debugview) when running a nightly build of QGIS 3.11 on Windows while the error happens?

**Comment by noitul** [comment]
> @noitul Could you copy any relevant logs from [DebugView](https://docs.microsoft.com/en-us/sysinternals/downloads/debugview) when running a nightly build of QGIS 3.11 on Windows while the error happens?

Hi letmaik,
I have another problem. I uninstalled all QGIS then installed "QGIS-OSGeo4W-3.11.0-46-Setup-x86_64.exe". After installing GIS4WRF plugin, the plugin does not work. Please find the Debugview log file attached. Thank you.
 
[qgis_gis4wrf.log](https://github.com/GIS4WRF/gis4wrf/files/3919219/qgis_gis4wrf.log)

**Comment by letmaik** [comment]
@noitul The log unfortunately doesn't contain all info. Can you copy-paste the error message that you see in QGIS? The end of the log says "See message log (Python Error) for more details" Maybe this helps.

**Comment by noitul** [comment]
Thank you for your help. Below is the first warning when QGISv3.11 is started.

2019-12-04T15:35:53     WARNING    warning:C:/QGIS/apps/qgis-dev/./python\qgis\utils.py:744: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
              mod = _builtin_import(name, globals, locals, fromlist, level)
             
             traceback: File "", line 1, in 
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 302, in loadPlugin
              __import__(packageName)
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\__init__.py", line 5, in 
              from gis4wrf.bootstrap import bootstrap
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\bootstrap.py", line 14, in 
              import pkg_resources
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\pkg_resources\__init__.py", line 78, in 
              from pkg_resources.extern import appdirs
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 1035, in _handle_fromlist
              File "", line 219, in _call_with_frames_removed
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 668, in _load_unlocked
              File "", line 638, in _load_backward_compatible
              File "C:\QGIS\apps\Python37\lib\site-packages\pkg_resources\extern\__init__.py", line 43, in load_module
              __import__(extant)
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\pkg_resources\_vendor\appdirs.py", line 560, in 
              import win32com.shell
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 953, in _find_and_load_unlocked
              File "", line 219, in _call_with_frames_removed
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\win32com\__init__.py", line 6, in 
              import pythoncom
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\pythoncom.py", line 2, in 
              import pywintypes
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\win32\lib\pywintypes.py", line 2, in 
              import imp, sys, os
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\imp.py", line 33, in 
              DeprecationWarning, stacklevel=2)
             
2019-12-04T15:35:53     WARNING    warning:C:\QGIS\apps\Python37\lib\site-packages\numpy\_distributor_init.py:32: UserWarning: loaded more than 1 DLL from .libs:
             C:\QGIS\apps\Python37\lib\site-packages\numpy\.libs\libopenblas.IPBC74C7KURV7CB2PKT5Z5FNR3SIBV4J.gfortran-win_amd64.dll
             C:\QGIS\apps\Python37\lib\site-packages\numpy\.libs\libopenblas.PYQHXLVVQ7VESDPUVUADXEVJOBGHJPAY.gfortran-win_amd64.dll
              stacklevel=1)
             
             traceback: File "", line 1, in 
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 354, in startPlugin
              if not _startPlugin(packageName):
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 334, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\__init__.py", line 24, in classFactory
              from gis4wrf.plugin.plugin import QGISPlugin
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\plugin.py", line 17, in 
              from gis4wrf.core import (
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\__init__.py", line 6, in 
              from gis4wrf.core.downloaders.dist import *
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\downloaders\dist.py", line 10, in 
              from gis4wrf.core.util import export, remove_dir
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\util.py", line 14, in 
              import numpy as np
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\numpy\__init__.py", line 140, in 
              from . import _distributor_init
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 1035, in _handle_fromlist
              File "", line 219, in _call_with_frames_removed
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\QGIS\apps\Python37\lib\site-packages\numpy\_distributor_init.py", line 32, in 
              stacklevel=1)
             
2019-12-04T15:35:54     WARNING    warning:C:\QGIS\apps\Python37\lib\importlib\_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
              return f(*args, **kwds)
             
             traceback: File "", line 1, in 
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 354, in startPlugin
              if not _startPlugin(packageName):
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 334, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\__init__.py", line 24, in classFactory
              from gis4wrf.plugin.plugin import QGISPlugin
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\plugin.py", line 17, in 
              from gis4wrf.core import (
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\__init__.py", line 16, in 
              from gis4wrf.core.readers.wrf_netcdf_metadata import *
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\readers\wrf_netcdf_metadata.py", line 7, in 
              import netCDF4 as nc
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\netCDF4\__init__.py", line 3, in 
              from ._netCDF4 import *
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 670, in _load_unlocked
              File "", line 583, in module_from_spec
              File "", line 1043, in create_module
              File "", line 219, in _call_with_frames_removed
             
2019-12-04T15:35:54     WARNING    warning:C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\constants.py:19: DeprecationWarning: np.asscalar(a) is deprecated since NumPy v1.16, use a.item() instead
              setattr(Constants, key.upper(), np.asscalar(val))
             
             traceback: File "", line 1, in 
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 354, in startPlugin
              if not _startPlugin(packageName):
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 334, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\__init__.py", line 24, in classFactory
              from gis4wrf.plugin.plugin import QGISPlugin
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\plugin.py", line 17, in 
              from gis4wrf.core import (
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\__init__.py", line 27, in 
              from gis4wrf.core.transforms.wrf_netcdf_to_gdal import *
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\transforms\wrf_netcdf_to_gdal.py", line 18, in 
              import wrf
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\__init__.py", line 18, in 
              from . import api
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 1035, in _handle_fromlist
              File "", line 219, in _call_with_frames_removed
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\api.py", line 6, in 
              from .constants import (ALL_TIMES, Constants, ConversionFactors,
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\constants.py", line 19, in 
              setattr(Constants, key.upper(), np.asscalar(val))
              File "", line 4, in asscalar
              File "C:\QGIS\apps\Python37\lib\site-packages\numpy\lib\type_check.py", line 546, in _asscalar_dispatcher
              'a.item() instead', DeprecationWarning, stacklevel=3)
             
2019-12-04T15:35:54     WARNING    warning:C:/QGIS/apps/qgis-dev/./python\qgis\utils.py:744: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
              mod = _builtin_import(name, globals, locals, fromlist, level)
             
             traceback: File "", line 1, in 
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 354, in startPlugin
              if not _startPlugin(packageName):
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 334, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\__init__.py", line 24, in classFactory
              from gis4wrf.plugin.plugin import QGISPlugin
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\plugin.py", line 17, in 
              from gis4wrf.core import (
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\__init__.py", line 27, in 
              from gis4wrf.core.transforms.wrf_netcdf_to_gdal import *
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\transforms\wrf_netcdf_to_gdal.py", line 18, in 
              import wrf
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\__init__.py", line 18, in 
              from . import api
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 1035, in _handle_fromlist
              File "", line 219, in _call_with_frames_removed
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\api.py", line 10, in 
              from .destag import destagger
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\destag.py", line 3, in 
              from .decorators import extract_and_transpose
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 983, in _find_and_load
              File "", line 967, in _find_and_load_unlocked
              File "", line 677, in _load_unlocked
              File "", line 728, in exec_module
              File "", line 219, in _call_with_frames_removed
              File "C:\Users\Administrator\AppData\Local\gis4wrf\python37\lib\site-packages\wrf\decorators.py", line 3, in 
              from collections import Iterable, OrderedDict
              File "C:/QGIS/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "", line 1032, in _handle_fromlist
              File "C:\QGIS\apps\Python37\lib\collections\__init__.py", line 52, in __getattr__
              DeprecationWarning, stacklevel=2)
             

Next is the error message when GIS4WRF plugin is activated

2019-12-04T15:38:00     WARNING    Traceback (most recent call last):
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\plugin.py", line 85, in show_dock
              self.dock_widget = MainDock(self.iface, self.dock_widget)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\dock.py", line 22, in __init__
              self.simulation_tab = SimulationTab(iface)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\tab_simulation.py", line 53, in __init__
              self.set_project_in_tabs()
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\tab_simulation.py", line 74, in set_project_in_tabs
              self.datasets_tab.project = project
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_datasets.py", line 71, in project
              self.populate_met_data_tree()
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_datasets.py", line 660, in populate_met_data_tree
              folder_meta, file_metas = read_grib_folder_metadata(time_range_folder)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\readers\grib_metadata.py", line 32, in read_grib_folder_metadata
              return read_grib_files_metadata(paths)
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\core\readers\grib_metadata.py", line 49, in read_grib_files_metadata
              assert meta.variables == variables
             AssertionError

**Comment by letmaik** [comment]
@noitul I created a separate issue for this error (#151) which is unrelated to the CRS issue here. To get into a workable state again, could you simply delete the met data folder which you think may have this issue and try again? It's probably one of the last ones you downloaded. The default data folder location is under Documents, e.g. Documents\gis4wrf\datasets\met\ds083.3\Analysis\201807151200-201807151800.

**Comment by noitul** [comment]
Thank you. The plugin loading problem was gone after removing ds083.3 folder. And the previous error appeared again. Below is the debugview log.

[9668] ..\..\..\src\core\qgsmessagelog.cpp(29) : (QgsMessageLog::logMessage) [7587ms] 2019-12-04T18:50:09 Messages[1] GIS4WRF : Path to WPS/WRF not set, functionality will be reduced. You can set the path under Settings > Options... > GIS4WRF
[9668] ..\..\..\src\core\qgsmessagelog.cpp(29) : (QgsMessageLog::logMessage) [9870ms] 2019-12-04T18:50:18 Python error[1] Traceback (most recent call last):
[9668]   File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_domains.py", line 313, in on_set_projection_button_clicked
[9668]     qgsProject.setCrs(get_qgis_crs(crs.proj4))
[9668]   File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\geo.py", line 28, in get_qgis_crs
[9668]     assert srs_id != -1
[9668] AssertionError
[9668] 
[9668] ..\..\..\src\core\qgsmessagelog.cpp(29) : (QgsMessageLog::logMessage) [0ms] 2019-12-04T18:50:18 Messages[1] Python error : An error has occurred while executing Python code: See message log (Python Error) for more details.

And this is from the massage log.

2019-12-04T18:50:18     WARNING    Traceback (most recent call last):
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\ui\widget_domains.py", line 313, in on_set_projection_button_clicked
              qgsProject.setCrs(get_qgis_crs(crs.proj4))
              File "C:/Users/Administrator/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\gis4wrf\plugin\geo.py", line 28, in get_qgis_crs
              assert srs_id != -1
             AssertionError 


Thank you for your time.

**Comment by letmaik** [comment]
@noitul Can you confirm that the log output is from QGIS 3.11 nightly build?

**Comment by noitul** [comment]
I used the install package from
https://qgis.org/downloads/weekly/QGIS-OSGeo4W-3.11.0-48-Setup-x86_64.exe.

**Comment by noitul** [comment]
Hi,
1. I uninstalled all QGIS programs then installed QGIS-OSGeo4W-3.4.10-1-Setup-x86_64.exe. But the same problem appeared.
2, I did step 1 again. But before installing QGIS 3.4.10, I deleted C:\Users\<user>I\AppData\Roaming\QGIS and C:\Users\<user>I\AppData\Local\QGIS. Then GIS4WRF worked.
3. Then I installed QGIS-OSGeo4W-3.10.0-2-Setup-x86_64.exe. In this case, the problem appeared again both on QGIS3.10.0 and QGIS3.4.10.

**Comment by letmaik** [comment]
I can successfully reproduce this issue now. Thanks everyone for pushing, will investigate and fix this soon.

**Comment by gsever** [comment]
@letmaik, just out of curiosity, what is the root of this problem? windows, QGIS, python, or any conflict with system packages? I am looking forward to using a newer QGIS and GIS4WRF on Windows.
Thanks for your efforts.

**Comment by letmaik** [comment]
@gsever I'm not sure yet what the underlying issue is, currently my best bet is it's a QGIS bug of a newer version (I reported it here: https://github.com/qgis/QGIS/issues/33519). The reason I couldn't repro it earlier was that the bug would only appear if the custom CRS did not exist yet in QGIS's database, and in my case I already had all the CRSs created from earlier 3.x versions. The databases are shared between 3.x installations/updates. I did a complete cleanup of QGIS and reinstalled it and then could reproduce the issue.

I think for now the only work-around is to use QGIS 3.4 (possibly 3.6, haven't tried) and clean the databases first, as @noitul described. On Windows this would be deleting the C:\Users<user>I\AppData\Roaming\QGIS and C:\Users<user>I\AppData\Local\QGIS folders.

**Comment by letmaik** [comment]
It looks very much like this is a bug with how QGIS databases are initialized, and it can happen in all versions at least down to 3.4 under certain conditions that are not entirely clear to me. The issue has been fixed 10 Dec 2019 (https://github.com/qgis/QGIS/pull/33328) and the patch backported to all major QGIS versions. It will be part of 3.10.2 (and equivalent earlier 3.x patch releases) which is not released yet. Related bug reports for the same issue: https://github.com/qgis/QGIS/issues/33038 https://github.com/qgis/QGIS/issues/32866
I will leave this issue open until we confirmed that the issue doesn't appear anymore in the new patch release.

**Comment by dmey** [comment]
Just a couple of updates for users wishing to use the GIS4WRF now:

1. The plugin still needs to be approved by QGIS before it becomes available at https://plugins.qgis.org/plugins/gis4wrf/ therefore users wishing to use the plugin in now can install it manually using the following zip file: [gis4wrf.zip](https://github.com/GIS4WRF/gis4wrf/files/4005469/gis4wrf.zip). I will let you know once this changes. 

2. I can confirm that deleting QGIS's profile folders (on `Windows: C:\Users\<user>\AppData\Roaming\QGIS` and `C:\Users\<user>\AppData\Local\QGIS`) and installing QGIS 3.6 (find it at https://qgis.org/downloads/) with the latest version of GIS4WRF worked with me -- so users may use this workaround until QGIS fixes the issue.

**Comment by gsever** [comment]
QGIS 3.6 (through QGIS-OSGeo4W-3.6.3-1-Setup-x86_64) and the latest gis4wrf.zip combo works for me here. No assertion errors while setting map CRS or any issues placing / browsing domain boxes.
Thank you for fixing this issue.

**Comment by dmey** [comment]
In QGIS 3.10.2 this issue seem to have been resolved which is available for download -- please upgrade to the latest one. Tested on Windows 10. I am going to close this issue now but please reopen if you encounter the same issue.

**Comment by gsever** [comment]
I verify that GIS4WRF (v0.14.2) works fine in QGIS 3.10.2-A on Windows 7. Thanks for your efforts again.
